source/code/providers/support/mycnf-scanner.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * --------------------------------- START OF LICENSE ----------------------------
+ *
+ * MySQL cimprov ver. 1.0
+ *
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved. 
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the ""Software""), to deal
+ * in the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * ---------------------------------- END OF LICENSE -----------------------------
+*/
+/**
+    \file        mycnf-scanner.cpp
+
+    \brief       Scans mysql configuration to extract authentication information.
+
+    \date        2014-11-05 16:22:00
+
+*/
+/*----------------------------------------------------------------------------*/
+
+#include "mycnf-scanner.h"
+
+namespace MySQL
+{
+    static inline bool GetPathFromDirective(const std::wstring& line, SCXCoreLib::SCXFilePath& path)
+    {
+        std::vector<std::wstring> elements;
+        SCXCoreLib::StrTokenize(line, elements, L" ");
+        if ( elements.size() != 2 )
+        {
+            return false;
+        }
+
+        path.Set(elements[1]);
+        return true;
+    }
+
+    bool MySQL_ConfigurationScanner::LoadCnfFile(
+        const SCXCoreLib::SCXFilePath& file, 
+        MySQL_Authentication& auth,
+        std::set< std::wstring >& visited, 
+        std::map<unsigned int, std::wstring>& ports)
```

Comment:
Well, I'm not sure I'm comfortable with this.

I think I'm right here for how WE designed the code. However, you're taking this over. In the end, it's you that's going to maintain this and support it going forward. Given that, you should have decision making authority for how the code is designed as long as it works properly and meets product requirements.

I feel comfortable in saying that putting implementation details in the object implementation makes perfect sense, and I view this as "object oriented design". It allows you to completely change the implementation of the object without ever changing the caller (a good thing).

But that said: You need to maintain the code. Given that, you should be comfortable with the code base and making decisions on your own.

Please discuss this with your team mates. In the end, I want you to do what you think is right.

My reviews are to catch some design errors (and there were some regarding privileges) and otherwise confirm to our coding standards. However, they are OUR coding standards (for my group), and may not apply to your group.

Please discuss with your team mates. This is a style issue and, if you really feel I'm wrong, do it as you wish. It won't affect product design.